extensions/amp-ad/0.1/amp-ad-xorigin-iframe-handler.js

```diff
@@ -39,6 +40,8 @@ export class AmpAdXOriginIframeHandler {
    * @param {!./amp-ad-3p-impl.AmpAd3PImpl|!../../amp-a4a/0.1/amp-a4a.AmpA4A} baseInstance
    */
   constructor(baseInstance) {
+    /** {!Window}*/
+    this.win = window;
```

Comment:
Yes, indeed, `this.baseInstance_.win` is the way to go. Yaiihh, for composition over inheritance :)